### ISSUE TITLE
Support structures vary depending on build plate position

### ISSUE THREAD

**Issue by evilC** [Ultimaker/Cura#7643]
**Application version**
4.5.0

**Platform**
Windows 10 x64  
nVidia Geforce 1080 Ti

**Printer**
Ender 3

**Reproduction steps**
1. Place model on build plate that has overhangs
2. Enable support
3. Slice
4. Translate model

**Screenshot(s)**
Same model, same layer, same settings, same rotation, but different X/Y position on build plate
![](https://i.imgur.com/AKBw3kc.png)  

![](https://i.imgur.com/ov8St7p.png)

**Actual results**
Support structure varies depending on build plate X/Y position

**Expected results**
Support structure should be identical regardless of X/Y position

**Project file**
[Brawler Cura bug.zip](https://github.com/Ultimaker/Cura/files/4564889/Brawler.Cura.bug.zip)

**Comment by Ghostkeeper** [comment]
Support is not specific to a particular model though. Take for instance this print here:

![image](https://user-images.githubusercontent.com/2448634/81237168-32cdc500-8fff-11ea-909e-609463fab84f.png)

If I move the selected object forward, which model would you expect the support to move along with?

**Comment by evilC** [comment]
One STL can contain multiple completely separate objects - this is one STL
![](https://i.imgur.com/ZfkVXJC.png)  
So when you slice, combine everything on the build plate to one logical object, and then calculate the origin point of the supports as the center of this one logical object.
That way, no matter where you place things on the build plate, as long as the relative offsets between them are the same, it will generate identical support

**Comment by Ghostkeeper** [comment]
Yeah, that would work.

Maybe it would be better for people that want to move their entire scene to the side of the printer for some reason.

**Comment by evilC** [comment]
Well it just seems a little silly as-is
You do a test print, ensure that the supports are sufficient, then add a duplicate of the print to the build plate (Which are nowhere near each other) but you don't get the same supports.
Also, it seems to me that the use-case you posted (two separate models very close together) is the edge-case here (Hell, I don't even know how to get pyramid-like support structures like that - the default options would make those two objects have completely unconnected support structures surely) and the software should not be geared towards that use-case to the detriment of the vastly more common use-case
TLDR multiple identical items on the build plate which are reasonably spaced should have **absolutely identical** support structures by default, anything else seems highly illogical

**Comment by mahtDFR** [comment]
We have discussed this and decided to defer it for now due to other priorities.
